javascript/435-Non-overlapping-Intervals.js

```diff
@@ -1,17 +1,32 @@
-let eraseOverlapIntervals = function(intervals) {
-    intervals = intervals.sort((a, b) => a[0] - b[1]);
+/**
+ * @param {number[][]} intervals
+ * Time O(N * logN) | Space O(1)
+ * @return {number}
+ */
+ var eraseOverlapIntervals = function(intervals) {
+    intervals.sort(([ aStart, aEnd ], [ bStart, bEnd ]) => aEnd !== bEnd
+        ? aEnd - bEnd
+        : aStart - bStart
+    );
 
-    let currentEnd = intervals[0][1];
-    let res = 0;
+    const length = intervals.length;
```

Comment:
nit: generally don't prefer extracting a single property to its own variable if it's only used once. prefer: `return intervals.length - gaps;`